File: realtime-server/src/schema/DungeonState.ts
import { Schema, MapSchema, type } from "@colyseus/schema";

export class Player extends Schema {
  @type("string") characterId: string = "";
  @type("float32") x: number = 0;
  @type("float32") z: number = 0;
  @type("float32") rotY: number = 0;
  @type("uint32") lastSeq: number = 0;
  @type("int32") hp: number = 100;
  @type("int32") resource: number = 0;
}

export class DungeonState extends Schema {
  @type({ map: Player }) players = new MapSchema<Player>();
  @type("uint32") tick: number = 0;
}

File: game-client/rise-after-dawn/Assets/Scripts/States/DungeonState.cs
using Colyseus.Schema;

namespace RiseAfterDawn.Client.States
{
    public partial class PlayerState : Schema
    {
        [Type(0, "string")]
        public string characterId = string.Empty;

        [Type(1, "float32")]
        public float x;

        [Type(2, "float32")]
        public float z;

        [Type(3, "float32")]
        public float rotY;

        [Type(4, "uint32")]
        public uint lastSeq;

        [Type(5, "int32")]
        public int hp = 100;

        [Type(6, "int32")]
        public int resource;
    }

    public partial class DungeonState : Schema
    {
        [Type(0, "map", typeof(MapSchema<PlayerState>))]
        public MapSchema<PlayerState> players = new MapSchema<PlayerState>();

        [Type(1, "uint32")]
        public uint tick;
    }
}
